src/Sarif.WorkItemFiling/OneResultPerWorkItemGroupingStrategy.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItemFiling
+{
+    /// <summary>
+    /// A grouping strategy that files each SARIF result as a separate work item.
+    /// </summary>
+    public class OneResultPerWorkItemGroupingStrategy : IGroupingStrategy
```

Comment:
>OneResultPerWorkItemGroupingStrategy [](start = 17, length = 36)

OMG. i loved this name when i was reviewing earlier!

`one result` per `workitem` grouping strategy - yay! so clear!

and then a few hours later - i came back to it and woah!!

the second time i read it as:

 `One result` per `workitemGrouping` strategy  -- which in my mind meant - one result for one work item group ---- which is always true isnt it?

SO.. 

may i recommmend we name this as: `OneResultPerWorkItemStrategy`

Reply:
I hear you, but since it is a "grouping strategy", I'd like to stick with this name.

---
In reply to: [306064261](https://github.com/microsoft/sarif-sdk/pull/1580#discussion_r306064261) [](ancestors = 306064261)